MoreLinq.Test/OrderedMergeTest.cs

```diff
@@ -0,0 +1,398 @@
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace MoreLinq.Test
+{
+    [TestFixture]
+    class OrderedMergeTest
+    {
+        [Test]
+        public void ShouldBeLazy()
+        {
+            var first = new BreakingSequence<object>();
+            var second = new BreakingSequence<object>();
+            Assert.DoesNotThrow(() => MoreEnumerable.OrderedMerge(first, second, id => id, id => id, id => id, id => id, (f, _) => f, null));
+        }
+
+        [Test]
+        public void ShouldDisposeEnumerators()
+        {
+            using (var first = TestingSequence.Of(new int[] { }))
+            using (var second = TestingSequence.Of(new int[] { }))
+                MoreEnumerable.OrderedMerge(first, second, id => id, id => id, id => id, id => id, (f, _) => f, null).ToArray();
+        }
+
+        [Test]
+        public void IfThereAreNoMoreElementsToReturnFromTheFirstCollectionThenReturnTheRemainingSecondCollection()
+        {
+            var first = new int[] { };
+            var second = new[] { 1, 2, 3 };
+
+            int Id(int id) => id;
```

Comment:
All local `Id` functions seem to be redundant and can be replace with a single non-local and generic and static method of this class:

```c#
static T Id<T>(T id) => id;
```